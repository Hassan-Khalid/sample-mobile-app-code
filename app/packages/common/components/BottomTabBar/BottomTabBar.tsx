import React from 'react';
import { FlatList, Image, TouchableOpacity, View } from 'react-native';
import styles from './styles/BottomTabBar.styles';
import Text from "../Text/Text"
import useBottomTabBar, { BottomTabs } from 'common/components/BottomTabBar/hooks/useBottomTabBar';
import { useSelector } from 'react-redux';
import { bottomTabSelector } from 'common/redux/selector/userSelector';



function BottomTabBar () {
 const {setCurrentTab} = useBottomTabBar()
  const  bottomTab  = useSelector(bottomTabSelector);
  
  
  const TabsView = ({item})=>{
  
    const {title,icon} = BottomTabs[item]
    const isFocused =  item === bottomTab ? icon.focused : icon.unFocused
    const textStyle = item === bottomTab ? styles.focusedText : styles.unFocusedText
    return(
      <View style={styles.tab}>
        <TouchableOpacity style={styles.tabInnerView} onPress={()=>setCurrentTab(item)} activeOpacity={0.8}>
          <Image source={isFocused} resizeMode={'contain'}/>
          <Text style={textStyle}>{title}</Text>
        </TouchableOpacity>
      </View>
    )
  }
  
  return (
    <View style={styles.container}>
      <FlatList
        scrollEnabled={false}
        contentContainerStyle={styles.tabContainer}
        data={Object.keys(BottomTabs)}
        renderItem={TabsView}/>
    </View>
  )
};

export default BottomTabBar;