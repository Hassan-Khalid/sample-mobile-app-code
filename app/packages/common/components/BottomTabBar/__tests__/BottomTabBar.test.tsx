import React from 'react';
import { render } from '@testing-library/react-native';
import BottomTabBar from '../BottomTabBar';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));


describe('BottomTabBar component', () => {
  it('renders correctly', () => {
    // Mocking useSelector hook
    // useSelector.mockReturnValue('Home'); // Mocking selected tab
    
    const wrapper = render(<BottomTabBar />);
    
    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
});
