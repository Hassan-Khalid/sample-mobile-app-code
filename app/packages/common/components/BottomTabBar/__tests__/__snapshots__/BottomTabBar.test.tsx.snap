// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomTabBar component renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": {
        "dynamic": {
          "dark": "#FFF",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#FFFFFF",
        },
      },
      "borderTopColor": {
        "dynamic": {
          "dark": "#25292c",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#e9edf1",
        },
      },
      "borderTopWidth": 1,
      "bottom": 1,
      "height": 82,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "width": 750,
      "zIndex": 999999,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
    data={
      [
        "WORK_ORDER",
        "MY_WORK_ORDER",
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignContent": "center",
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "testUri": "../../../packages/common/assets/Images/common/ic_WO_grey.png",
                }
              }
            />
            <Text
              style={
                {
                  "color": {
                    "dynamic": {
                      "dark": "#748094",
                      "highContrastDark": undefined,
                      "highContrastLight": undefined,
                      "light": "#8798ad",
                    },
                  },
                  "fontSize": 14,
                  "margin": 4,
                }
              }
            >
              tickets.closedTicketsPage.totalTickets
            </Text>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignContent": "center",
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "testUri": "../../../packages/common/assets/Images/common/ic_myWO_grey.png",
                }
              }
            />
            <Text
              style={
                {
                  "color": {
                    "dynamic": {
                      "dark": "#748094",
                      "highContrastDark": undefined,
                      "highContrastLight": undefined,
                      "light": "#8798ad",
                    },
                  },
                  "fontSize": 14,
                  "margin": 4,
                }
              }
            >
              tickets.myTicketsPage.openTicketsTitle
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
